tools.py

```diff
@@ -143,13 +149,83 @@ def esn_prediction(data, params):
               spectral_radius=params['rho'],
               noise=params['noise'])
 
-    trainlen = params['trainlen']
+    # train the ESN
+    prediction = np.ones((futureTotal, n_vars))
+    window_pred = np.ones((window, n_vars))
+
+    for i in range(0, futureTotal, window):
+        data_slice = data[-trainlen - futureTotal + i:-futureTotal + i]
+        pred_training = esn.fit(np.ones((trainlen, n_vars)),
+                                data_slice)
+        inter_pred = esn.predict(window_pred)
+        prediction[i:i + window] = inter_pred
```

Comment:
I handle pep8 by using an auto-linter called ``autopep8``. 

You can try it yourself: 
```bash
pip install autopep8
autopep8 --in-place --aggressive --aggressive <myfile>.py
```
This will automatically fix most pep8 issues (if it can). Sometimes can't handle lines that are too long, automatically.

Reply:
I was looking on pep8.org, and they have
`ham[lower+offset : upper+offset]` should be used over
`ham[lower + offset:upper + offset]`.
`ham[lower + offset : upper + offset]` is also good.
I may be misunderstanding, but it seems like this is something autopep8 should change. What do you think?